codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -232,12 +237,17 @@ abstract class ConcurrentWriterCpgPass[T <: AnyRef](cpg: Cpg, outName: String =
     }
   }
 
-  private class Writer(serializedCpg: SerializedCpg, prefix: String, inverse: Boolean) extends Runnable {
+  private class Writer(serializedCpg: SerializedCpg,
+                       prefix: String,
+                       inverse: Boolean,
+                       mdc: java.util.Map[String, String] = MDC.getCopyOfContextMap)
+      extends Runnable {
 
     val queue = new LinkedBlockingQueue[Option[DiffGraph]](ConcurrentWriterCpgPass.writerQueueCapacity)
 
     override def run(): Unit = {
       try {
+        MDC.setContextMap(mdc)
```

Comment:
Did we not set this in line 40?

Reply:
It may look like it, but their are two entirely different inner `private class Writer` :(